### PR TITLE
Fix translation of undefined function call when debug info is present

### PR DESCRIPTION
**Pull request by vmaksimo**
This fix resolves issue:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/524

Function translation invokes translation of LLVM loop metadata as a last step.
SPIRVToLLVM keeps a map with metadata. If function with loop metadata is
forward-referenced, translation starts and metadata is placed in the map. If
this function contains calls of undefined functions inside, it will fail -
metadata map is not empty, but there is no loop info inside the undefined
function.

Example:
```
float bar(int x);

__kernel void foo(__global float* outPtr, int i) {
  for (int j = 0; j < i; ++j) {
    outPtr[j] = bar(j);
  }
}
```

### CODE REVIEW

File: test/DebugInfo/Generic/undef-func-call.ll [hunk 1]
```diff
@@ -0,0 +1,98 @@
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o - | llvm-dis -o %t.ll
+; RUN: llc -mtriple=%triple < %t.ll
```
Comment (AlexeySotkin) [inline test/DebugInfo/Generic/undef-func-call.ll:4]: Why do you run llc?

File: test/DebugInfo/Generic/undef-func-call.ll [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
```
Comment (AlexeySotkin) [inline test/DebugInfo/Generic/undef-func-call.ll:5]: I think the tests should have CHECK lines, or an explicit comment that this is a regression test for crash reported in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/524

Comment (vmaksimo) [reply test/DebugInfo/Generic/undef-func-call.ll:5]: Actually I just wanted to check that translator doesn't crash in this pipeline. Will add a comment (and remove llc run as a redundant run)

File: test/DebugInfo/Generic/undef-func-call.ll [hunk 3]
```diff
@@ -0,0 +1,100 @@
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o - | llvm-dis -o %t.ll
+
+; This is a regression test for reported issue https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/524.
+; Test checks that reverse translation will not fail.
```
Comment (AlexeySotkin) [inline test/DebugInfo/Generic/undef-func-call.ll:6]:
```suggestion
; Test checks that reverse translation will not fail with assertion.
```

File: test/DebugInfo/Generic/undef-func-call.ll [hunk 4]
```diff
@@ -0,0 +1,100 @@
+; RUN: llvm-as < %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o - | llvm-dis -o %t.ll
```
Comment (AlexeySotkin) [inline test/DebugInfo/Generic/undef-func-call.ll:3]: I think `llvm-dis` is not needed as well.

**Comment by vmaksimo** [comment]
> Please squash the commits.

Can it be "squash and merged" automatically when merging PR?

**Pull request merged**